client/src/js/services/ProjectService.js

```diff
@@ -1,15 +1,16 @@
 angular.module('bhima.services')
 .service('ProjectService', ProjectService);
 
-ProjectService.$inject = [ '$http', 'util' ];
+ProjectService.$inject = [ '$http', '$translate', 'util' ];
 
-function ProjectService($http, util) {
+function ProjectService($http, $translate, util) {
   var service = this;
 
   service.create = create;
   service.read = read;
   service.update = update;
-  service.delete = del;
+  service.del = del;
+  service.readComplete = readComplete;
```

Comment:
You can use the [$http](https://docs.angularjs.org/api/ng/service/$http) service to set querystrings for you via a configuration object.  [Here is the relevant documenation](https://docs.angularjs.org/api/ng/service/$http#get).

Instead of creating a whole new method, you should just use `service.read`!  Something like this:
```js
function read(id, params) {
   var url = (id) ? '/projects/' + id : '/projects';
   return $http.get(url, { params : params });
}

// in your controller
ProjectService.read(null, { complete : 1 })
.then(function (projects) {
  console.log('The complete projects:', projects);
});
```
AngularJS should automatically serialize the parameters into a query string for you.